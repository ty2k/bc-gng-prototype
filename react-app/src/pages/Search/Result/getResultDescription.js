export default function getResultDescription(result, tab) {
  // Documents (3) tab
  if (tab === 3) {
    return (
      <p
        className="description"
        dangerouslySetInnerHTML={{ __html: result?.S?.toString() }}
      />
    );

    // All (1), Services (4), News (2) tabs and default
  } else {
    return (
      <p className="description">
        {result?.MT?.length > 0 &&
          result?.MT?.map((metaTag) => {
            if (metaTag?.$?.N === "description") {
              return metaTag?.$?.V;
            }
            // .map() expects an explicit return value and React will
            // throw a warning if this is not present
            return null;
          })}
      </p>
    );
  }
}
