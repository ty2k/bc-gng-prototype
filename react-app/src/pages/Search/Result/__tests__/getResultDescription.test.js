import TestRenderer from "react-test-renderer";

import getResultDescription from "../getResultDescription";

import {
  mockResultTab1,
  mockResultTab2,
  mockResultTab3,
  mockResultTab4,
} from "../__mocks__/mockResultData";

describe("getResultDescription", () => {
  const descriptionTab1 = TestRenderer.create(
    getResultDescription(mockResultTab1, 1)
  ).toJSON();
  const descriptionTab2 = TestRenderer.create(
    getResultDescription(mockResultTab2, 2)
  ).toJSON();
  const descriptionTab3 = TestRenderer.create(
    getResultDescription(mockResultTab3, 3)
  ).toJSON();
  const descriptionTab4 = TestRenderer.create(
    getResultDescription(mockResultTab4, 4)
  ).toJSON();

  it("renders a Tab 1 (All) description correctly", () => {
    expect(descriptionTab1).toMatchSnapshot();
  });
  it("renders a Tab 2 (News) description correctly", () => {
    expect(descriptionTab2).toMatchSnapshot();
  });
  it("renders a Tab 3 (Documents) description correctly", () => {
    expect(descriptionTab3).toMatchSnapshot();
  });
  it("renders a Tab 4 (Services) description correctly", () => {
    expect(descriptionTab4).toMatchSnapshot();
  });
});
