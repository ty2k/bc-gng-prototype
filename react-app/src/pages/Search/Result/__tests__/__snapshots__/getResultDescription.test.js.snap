// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getResultDescription renders a Tab 1 (All) description correctly 1`] = `
<p
  className="description"
>
  Information on both included and excluded benefits eligibility.
</p>
`;

exports[`getResultDescription renders a Tab 2 (News) description correctly 1`] = `
<p
  className="description"
>
  As the Province eliminates Medical Service Plan (MSP) premiums for British Columbians, an updated payment method will ensure international students continue to contribute to, and benefit from, B.C.â€˜s health-care coverage.
</p>
`;

exports[`getResultDescription renders a Tab 3 (Documents) description correctly 1`] = `
<p
  className="description"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<b>MSP</b> Service Locations Codes...1, 2021, the <b>MSP</b> Service Locations Codes...where publicly funded <b>MSP</b> services are provided...or tracked through
<b>MSP</b>.

(Q) Specialist Physician",
    }
  }
/>
`;

exports[`getResultDescription renders a Tab 4 (Services) description correctly 1`] = `
<p
  className="description"
>
  Medical Services Plan for B.C. Residents
</p>
`;
